jdat_notebooks/MRS_Mstar_analysis/JWST_Mstar_dataAnalysis_PointSourceDetectorBasedExtraction.ipynb

```diff
@@ -0,0 +1,2258 @@
+{
```

Comment:
<p>1. Please move the <code>tarfile</code> import to the top with the others.</p><p> </p><p>2. Gathering the names of these tar files into a list and looping through it would avoid the need to re-copy the filenames and similar code, which will help make future maintenance easier.</p><p> </p><p>Here is an approach you could use to condense this cell and the next one:</p><p> </p><p><code>from pathlib import Path # would move to top with other imports</code></p><p><code> </code></p><p><code>tar_location = Path('.')</code></p><p><code>tar_files = ['file1.tar.gz', 'file2.tar.gz'] # and so on</code></p><p><code>tar_files_full = [tar_location / f for f in tar_files]</code></p><p><code>tar_urls = ['https://data.science.stsci.edu/file1',</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;'https://data.science.stsci.edu/file2'] # and so on</code></p><p><code> </code></p><p><code>for p, url in zip(tar_paths_full, tar_urls):</code></p><p><code>&nbsp;&nbsp;&nbsp;# check for un-tarred version of file</code></p><p><code>&nbsp;&nbsp;&nbsp;if os.path.exists(p.parent / p.stem):</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print(f"{p.name} exists.")</code></p><p><code>&nbsp;&nbsp;&nbsp;else:</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# check for tarred version of file and un-tar</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if os.path.exists(p):</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;tar = tarfile.open(p, "r:gz")</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;tar.extractall()</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;tar.close()</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# otherwise, download and un-tar file</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;else:</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;urllib.request.urlretrieve(url, p)</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print(f"Downloading {p.name}...")</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/dat_pyinthesky/pull/162/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Reply:
Here's a clearer version of the code block:
```
from pathlib import Path # would move to top with other imports

tar_location = Path('.')
tar_files = ['file1.tar.gz', 'file2.tar.gz'] # and so on
tar_files_full = [tar_location / f for f in tar_files]
tar_urls = ['https://data.science.stsci.edu/file1',
            'https://data.science.stsci.edu/file2'] # and so on

for p, url in zip(tar_paths_full, tar_urls):
    # check for un-tarred version of file
    if os.path.exists(p.parent / p.stem):
        print(f"{p.name} exists.")
    else:
        # check for tarred version of file and un-tar
        if os.path.exists(p):
            tar = tarfile.open(p, "r:gz")
            tar.extractall()
            tar.close()
        # otherwise, download and un-tar file
        else:
            urllib.request.urlretrieve(url, p)
            print(f"Downloading {p.name}...")
```

Reply:
done

Reply:
This is not done; the `tarfile` import is now missing from the notebook.